ledger/catchpointtracker.go

```diff
@@ -0,0 +1,908 @@
+		} else {
+			ct.catchpointInterval = 0
+		}
+	case 1:
+		ct.catchpointInterval = cfg.CatchpointInterval
+	}
+
+	ct.catchpointFileHistoryLength = cfg.CatchpointFileHistoryLength
+	if cfg.CatchpointFileHistoryLength < -1 {
+		ct.catchpointFileHistoryLength = -1
+	}
+}
+
+// GetLastCatchpointLabel retrieves the last catchpoint label that was stored to the database.
+func (ct *catchpointTracker) GetLastCatchpointLabel() string {
+	ct.catchpointsMu.RLock()
+	defer ct.catchpointsMu.RUnlock()
+	return ct.lastCatchpointLabel
+}
+
+// loadFromDisk loads the state of a tracker from persistent
+// storage.  The ledger argument allows loadFromDisk to load
+// blocks from the database, or access its own state.  The
+// ledgerForTracker interface abstracts away the details of
+// ledger internals so that individual trackers can be tested
+// in isolation.
+func (ct *catchpointTracker) loadFromDisk(l ledgerForTracker, lastBalancesRound basics.Round) (err error) {
+	ct.log = l.trackerLog()
+	ct.dbs = l.trackerDB()
+
+	err = ct.dbs.Wdb.Atomic(func(ctx context.Context, tx *sql.Tx) error {
+		err0 := ct.accountsInitializeHashes(ctx, tx, lastBalancesRound)
+		if err0 != nil {
+			return err0
+		}
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	ct.accountsq, err = accountsInitDbQueries(ct.dbs.Rdb.Handle, ct.dbs.Wdb.Handle)
+	if err != nil {
+		return
+	}
+
+	ct.lastCatchpointLabel, _, err = ct.accountsq.readCatchpointStateString(context.Background(), catchpointStateLastCatchpoint)
+	if err != nil {
+		return
+	}
+
+	writingCatchpointRound, _, err := ct.accountsq.readCatchpointStateUint64(context.Background(), catchpointStateWritingCatchpoint)
+	if err != nil {
+		return err
+	}
+	if writingCatchpointRound == 0 || !ct.catchpointEnabled() {
+		return nil
+	}
+	var dbRound basics.Round
+	// make sure that the database is at the desired round.
+	err = ct.dbs.Rdb.Atomic(func(ctx context.Context, tx *sql.Tx) (err error) {
+		dbRound, err = accountsRound(tx)
+		return
+	})
+	if err != nil {
+		return err
+	}
+	if dbRound != basics.Round(writingCatchpointRound) {
+		return nil
+	}
+
+	blk, err := l.Block(dbRound)
+	if err != nil {
+		return err
+	}
+	blockHeaderDigest := blk.Digest()
+
+	ct.catchpointWriting = 0
+	// keep these channel closed if we're not generating catchpoint
+	ct.catchpointSlowWriting = make(chan struct{}, 1)
+	close(ct.catchpointSlowWriting)
+	ct.roundDigest = nil
+
+	ct.generateCatchpoint(context.Background(), basics.Round(writingCatchpointRound), ct.lastCatchpointLabel, blockHeaderDigest, time.Duration(0))
+	return nil
+}
+
+// newBlock informs the tracker of a new block from round
+// rnd and a given ledgercore.StateDelta as produced by BlockEvaluator.
+func (ct *catchpointTracker) newBlock(blk bookkeeping.Block, delta ledgercore.StateDelta) {
+	ct.catchpointsMu.Lock()
+	defer ct.catchpointsMu.Unlock()
+	ct.roundDigest = append(ct.roundDigest, blk.Digest())
+}
+
+// committedUpTo informs the tracker that the database has
```

Comment:
probably this large comment should live in interface definition and not in each tracker since it is generic. Plus, it does not mention `lookback`